client/src/app/service/auth.service.ts

```diff
@@ -0,0 +1,62 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { map } from 'rxjs/operators';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { environment } from 'src/environments/environment';
+
+export interface User {
```

Comment:
I am a fan of extracting every class and interface to separate files, but it not really necessary to do it :)